machine/machine_core/ssh_connection.py

```diff
@@ -221,14 +226,16 @@ def _kill_ssh_master(self):
             self.ssh_master = None
         if self.ssh_process:
             self.message("killing ssh master process", str(self.ssh_process.pid))
+            assert self.ssh_process.stdin, "no stdin for SSH master process"
```

Comment:
I'm not a fan of the commentary on `assert`s, and please use `is not None` here, if that's what you mean.